src/pb_stub.cc

```diff
@@ -837,6 +857,135 @@ Stub::GetOrCreateInstance()
   return Stub::stub_instance_;
 }
 
+void
+Stub::LaunchLogRequestThread()
+{
+  log_thread_ = true;
+  log_monitor_ = std::thread(&Stub::ServiceLogRequests, this);
+}
+
+void
+Stub::TerminateLogRequestThread()
+{
+  log_thread_ = false;
+  {
+    bi::scoped_lock<bi::interprocess_mutex> guard{log_message_mutex_};
+    log_request_buffer.push(DUMMY_MESSAGE);
+  }
+  log_message_cv_.notify_all();
```

Comment:
The notification must be sent when the mutex is acquired.

Reply:
Is this true? ref:https://en.cppreference.com/w/cpp/thread/condition_variable/notify_all

> The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock.

Reply:
I was following the convention in the above link. I attempted to place the notification within the scope of the lock and it ends up stalling the program.

Reply:
Thanks for the link Ryan. Looks like it is not required. Resolving this thread.